lib/fabrik/DirectorManager.js

```diff
@@ -240,6 +245,30 @@ class DirectorManager extends BaseManager {
       });
   }
 
+  executePreUpdateActions(deploymentName, context) {
+    _.assign(context, {
```

Comment:
You are not cloning context. So the same is passed to executeActions which will missout instance id in create flow, is that ok? I would rather not modify state of a pass by reference which is conditional. why not do this setting when actionContext is getting created, that way its consistent and same flows across.

Reply:
The idea behind adding extra setting to actionContext in executeActions was that it avoids duplication of settings in all the lifecycles which might be calling executeActions. But yeah, better to clone context as it is passed by reference (missed this part, thanks)